third_party/move/move-compiler/src/cfgir/optimize/inline_blocks.rs

```diff
@@ -131,9 +146,18 @@ fn remap_to_last_target(
     if remapping.is_empty() {
         return;
     }
-    // populate remapping for non changed blocks
+    // populate remapping for non changed blocks, and also
```

Comment:
Adding self mapping to every label that is not actually remapped seem wasteful. It looks like this may have been done so that "all labels are present in the remapping". Instead, `unwrap_or_` with the default may have sufficed (that is, when looking up a label in the remapping, absence indicates "use key as the value", and we get the same semantics), likely reducing the size of the mapping.

This is not a behavior introduced by this PR, so I don't expect this change to be made in this PR. I just wanted to bring this to attention, and see if what I say makes sense.

Reply:
I think this was done to simplify `remap_labels()` (defined in cfgir/ast.rs).  I'll try to rewrite that without making it too ugly.